code/modules/surgery/organs/eyes.dm

```diff
@@ -44,6 +47,29 @@
 	M.update_tint()
 	M.update_sight()
 
+/obj/item/organ/eyes/on_life()
+	..()
+	var/mob/living/carbon/C = owner
+	//since we can repair fully damaged eyes, check if healing has occurred
+	if(failing && (damage < maxHealth))
+		failing = FALSE
+		C.cure_blind(EYE_DAMAGE)
+	//various degrees of "oh fuck my eyes", from "point a laser at your eye" to "staring at the Sun" intensities
+	if(damage > 20)
+		damaged = TRUE
+		if(failing)
+			C.become_blind(EYE_DAMAGE)
+		else if(damage > 30)
+			C.overlay_fullscreen("eye_damage", /obj/screen/fullscreen/impaired, 2)
+		else
+			C.overlay_fullscreen("eye_damage", /obj/screen/fullscreen/impaired, 1)
+	//called once since we don't want to keep clearing the screen of eye damage for people who are below 20 damage
+	else if(damaged)
+		damaged = FALSE
+		C.clear_fullscreen("eye_damage")
+	return
```

Comment:
this was previously ran just when damaged.

Should this be on AdjustOrganDamage

Reply:
From what I saw it was never run, and after having removed it you could still get eye damage in the form of short/near-sightedness through welding and things like the screwdriver. This was moved here since I was going to do more cleanup with the organ types to have everything run off of organ damage, but abandoned that idea.

Reply:
Tested it again, and welding doesn't give eye damage, so it would seem I was incorrect. I'll try to fix this.

Reply:
Anime coding counterpart fixed it to some degree. Really should've left it alone, but I felt it necessary to move all organ_damage procs to the ones part of the organ itself. Welding worked as it did prior, as did stabbing someone in the eyes last I tested it.